int32 mode
---
float32 theta
